src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/index.jsx

```diff
@@ -15,12 +18,47 @@ import styled from 'react-emotion';
 import space from 'app/styles/space';
 import {withTheme} from 'emotion-theming';
 
+const api = new Client();
+
 class SentryApplicationRow extends React.PureComponent {
   static propTypes = {
     app: PropTypes.object.isRequired,
     orgId: PropTypes.string.isRequired,
   };
 
+  redirectUrl = install => {
+    const {orgId, app} = this.props;
+    let redirectUrl = `/settings/${orgId}/integrations/`;
+
+    if (app.redirectUrl) {
+      const url = parseurl({url: app.redirectUrl});
+      // Order the query params alphabetically.
+      const installQuery = JSON.parse(
+        JSON.stringify({installationId: install.uuid, code: install.code})
+      );
+      const query = Object.assign(qs.parse(url.query), installQuery);
+      redirectUrl = `${url.protocol}//${url.host}${url.pathname}?${qs.stringify(query)}`;
+    }
+
+    return redirectUrl;
+  };
+
+  install = () => {
+    const {orgId, app} = this.props;
+
+    const success = install => browserHistory.push(this.redirectUrl(install));
+    const error = err => addErrorMessage(err.responseJSON);
```

Comment:
should we have a successMessage for when people get redirected to `/settings/{orgId}/integrations`?

Reply:
oo, yea, good call.